### ISSUE TITLE
Interactivity warning / binder link / download link of notebooks

### ISSUE THREAD

**Issue by maximlt** [holoviz/holoviz#320]
Some documentation pages, that are built out of notebooks, require an active Python kernel to be fully interactive. For instance, anything that creates datashaded Bokeh plots would need Python running in the background to allow users to see that datashading happens every time they zoom in and out. Getting to a stage where the full interactivity would directly be available to users when they visit a site seems like it might one day be possible thanks to WASM, yet all of that is still pretty experimental and also comes with some disadvantages (e.g. increased page size). So for now we have to live with having no active Python kernel and static websites. To let the user know about this, a few solutions are available:

- a download link can be offered, which allows users to download the notebook and run it to get the full-interactivity. It assumes they have already set up their environment to run the notebook. Note that a download link is useful even if the users aren't interested in the full-interactivity, for instance if they just want to play around with the code, want that notebook as a starting point, etc.
- a Binder link can be offered. It has the same benefits as the download link, with the advantage of not forcing the user to install anything locally to run the notebook, and the disadvantage that Binder doesn't always work (it's a free service after all 🙃 )
- an interactivity warning box can be added to the page to indicate users that not all the interactivity will work in the page. This box can include download and binder links.

To illustrate these solutions, currently on this [Panel page](https://panel.holoviz.org/reference/panes/Audio.html#panes-gallery-audio) there's:

- a download link at the page start
- a Binder link in the right side bar
- an interactivity warning box in the right side bar, that offers a download link
- a download link at the page bottom (not shown in the screenshot)

<img width="1677" alt="image" src="https://user-images.githubusercontent.com/35924738/183913742-f91f1f99-2763-4768-a5ef-427ac3ecb986.png">

As can be seen, these solutions can each be improved individually, but also should probably be rearranged/merged so that we don't end up with as many links as on the Panel page above.

For now I'm not making any suggestion, I'm just opening the discussion :) I would also suggest not focusing too much on the tooling to start the discussion (e.g. what `nbsite` does now, what `pydata-sphinx-theme` offers, etc.), instead I believe the discussion should focus first on the user experience, and then seek whether the available tools offer what has been decided, and if not see how hard it would be to implement it ourselves.

**Comment by maximlt** [comment]
Just to document another way to let users know the page isn't fully interactive.

<img width="1602" alt="image" src="https://user-images.githubusercontent.com/35924738/184958822-4cccaef6-6c72-4a2b-beae-c622c4f95fad.png">

**Comment by jbednar** [comment]
I agree that this is a major mess and should be dealt with. Note that most people won't understand what "static websites" are, so I don't think that warning is effective. And any warning that's at a specific point in the page, not visible when scrolled off, won't be very effective, because people click around rather than reading pages straight through.

What I think would be more effective is some sort of watermark actually covering the figure with a message about it only being an example and not fully interactive. Such a message would be ugly enough that we would want to get rid of it anywhere that it's not actually needed, which I think would be a good practice for us, as it would motivate us to remove such limitations wherever possible.

**Comment by droumis** [comment]
I agree and see three problems: 
1. Some users expect full interactivity, don't see the warnings, and then get dissapointed by holoviz tools.
2. Some users see the warnings about needing to download for full interactively, and now have to go through the process of getting their environment set up, which adds cognitive and practical overhead.
3. The binder link is small and in a weird location under "on this page", so I think a lot of people miss it.

The ideal solution is of course to have them be fully interactive, which goes into an infrastructure and implementation discussion. I don't want to derail this conversation, but I think it would be worth discussing at some point soon.

For the near term, sticking with the current infrastructure, I think some sort of more prominent watermark or alert banner is an ugly but effective temporary idea. It doesn't have to cover the whole figure; maybe just the corner, or as a [banner that can be closed by the user](https://getbootstrap.com/docs/4.0/components/alerts/#:~:text=Holy%20guacamole!%20You%20should%20check%20in%20on%20some%20of%20those%20fields%20below.). Given that binder potentially solves the second problem (when it works) I think it should be merged and incorporated into the banner. Something like:

> "Follow [install instructions]() for full Python-backed interactivity, or [open this page in Binder]()

If people don't like the plot watermark/banner idea, I think at least adding the binder link/text to the sticky bottom right alert box that says "...Right click to download and run locally..." is an improvement.

**Comment by maximlt** [comment]
Do we have any data on the share of users who are reading the docs on mobile? The right sidebar isn't displayed on mobile so people will miss any warning that is in there. This might make the banner a better option on mobile, if it can be closed and isn't too big. It would be nice to have a small mock up of how that would look like on desktop and mobile.

**Comment by droumis** [comment]
Since Jan 1, 2022, the fraction of mobile users on our websites:

hvplot is at ~11%
panel is at ~18%

all the other sites that we are tracking are somewhere within that range

**Comment by maximlt** [comment]
Interesting thanks! That is higher that I thought and definitely not negligible.

**Comment by apowers313** [comment]
My experience (as a first time user) was "why the heck are these graphs partially interactive... on hover works, panning works, selection works... but none of the filters or settings work". Note that it's when I go to interact with the Panel that my expectations aren't met. If you can detect those interactions (onclick? onblur? special DemoPanel? I dunno) and pop up a warning like "you're about to download 40MB WASM binary" and then if they click "ok" run it... well, not sure it's the perfect CX but at least it's foolproof. :)

**Comment by jbednar** [comment]
Note that @apowers313 is discussing the Panel website originally, not the HoloViz website. The HoloViz website does not offer WASM-based interactivity at all because it relies on Datashader, which needs LLVM, which isn't yet available for WASM.
